### PR TITLE
feat: Add deploy workflow

### PR DESCRIPTION
**Pull request by joshlarson**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206954553660646

### CODE REVIEW

File: .github/workflows/deploy-ecs.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      environment:
+        type: environment
+        required: true
+        default: dev
+  push:
+    branches: ["main"]
+
+jobs:
+  call-workflow:
+    uses: mbta/workflows/.github/workflows/deploy-ecs.yml@v2
```
Comment (firestack) [inline .github/workflows/deploy-ecs.yml:15]: question(non-blocking): what benefit do we get from pinning this dependency to `@v2` if this is an internal resource?

Comment (firestack) [inline .github/workflows/deploy-ecs.yml:15]: question: do you know if dependabot will update this with our current configuration?

Comment (joshlarson) [reply .github/workflows/deploy-ecs.yml:15]: I don't think so - I don't think dependabot updates things defined in these workflows (at least not as it's configured in Skate - I updated a bunch of those by hand a month or so ago).

Comment (firestack) [reply .github/workflows/deploy-ecs.yml:15]: Well I think we should be able to have dependabot do this, but we can explore that later. https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot

**Review by firestack — APPROVED** [review]
I'd like a ticket or followup PR to [add dependabot for the workflows](https://github.com/mbta/ors-deploy/pull/2/files#r1552328910) but this looks good as is!

File: .github/workflows/deploy-ecs.yml (hunk 1 above)
Comment (joshlarson) [reply .github/workflows/deploy-ecs.yml:15]: [Ticket for this repo](https://app.asana.com/0/1148853526253420/1207005059291023/f)
[Ticket for Skate](https://app.asana.com/0/1148853526253420/1207005059291028/f)

**Comment by joshlarson** [comment]
I asked in #tid-infrastructure, just to confirm that this looks like it will connect with the associated devops PR. I'll merge once I get a happy answer there!

File: .github/workflows/deploy-ecs.yml (hunk 1 above)
Comment (firestack) [reply .github/workflows/deploy-ecs.yml:15]: Thank you!

Comment (krisrjohnson21) [reply .github/workflows/deploy-ecs.yml:15]: @firestack While we don't really own workflows, Infra is in favor of pinning versions. That way, it's easier for us to roll out changes in a measured and structured fashion, like we did when moving to assumed roles and away from long-lived access keys for workflows.

**Pull request merged**